File: two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::set<int> numsSet(nums.begin(), nums.end());
        std::vector<int> answer;

        for(int i = 0; i < nums.size(); i++) {
            int temp = target - nums[i];
            if(auto search = numsSet.find(temp); search != numsSet.end()) {
                auto tempIdx = std::find(nums.begin(), nums.end(), temp);
                int index = tempIdx - nums.begin();
                if(index != i) {
                    answer.push_back(i);
                    answer.push_back(index);
                    break;
                }
            }
        }
        return answer;
    }
};

File: day10/twoSum.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
  let l = 0, r = numbers.length - 1;
  while (l !== r) {
    let temp = numbers[l] + numbers[r];
    if (temp === target) {
      return [l + 1, r + 1];
    } else if (temp > target) {
      r--;
    } else {
      l++;
    }
  }
  return [];
};